awx/ui_next/src/screens/Job/JobOutput/JobOutput.jsx

```diff
@@ -131,9 +180,21 @@ class JobOutput extends Component {
   }
 
   componentWillUnmount() {
+    if (ws) {
+      ws.close();
+    }
+    clearInterval(this.interval);
     this._isMounted = false;
   }
 
+  monitorJobSocketCounter() {
+    const { remoteRowCount } = this.state;
+    if (this.jobSocketCounter >= remoteRowCount) {
+      this._isMounted &&
+        this.setState({ remoteRowCount: this.jobSocketCounter + 1 });
+    }
+  }
+
   async loadJobEvents() {
     const { job, type } = this.props;
 
```

Comment:
i really like how simple this PR is...and I like how you've handled the connection/dealing with ws management logic and the do stuff with data specific to the page part of things.